vignettes/generation.Rmd

```diff
@@ -0,0 +1,170 @@
+    "Normal2", "Normal1", "Normal", "Patient2", system.file("extdata", "NSCLC_MeDIP_2N_fst_chr_20_21_22.bam", package = "MEDIPSData", mustWork = TRUE),
+    "Tumour1", "Tumour1", "Tumour", "Patient1", system.file("extdata", "NSCLC_MeDIP_1T_fst_chr_20_21_22.bam", package = "MEDIPSData", mustWork = TRUE),
+    "Tumour2", "Tumour2", "Tumour", "Patient2", system.file("extdata", "NSCLC_MeDIP_2T_fst_chr_20_21_22.bam", package = "MEDIPSData", mustWork = TRUE)
+    )
+```
+
+## Making a qseaSet
+The `qsea` tutorial details the steps involved in generating a `qseaSet`. 
+We provide a function, `makeQset`, which wraps these up into one function, although you may still use the rest of the `mesa` functionality with or without using this to create your `qseaSet`.
+
+Before creating the `qseaSet`, you need to have:
+
+* The `sampleTable` of metadata, including the file paths to the bam files.
+* The genome to which the bam files have been mapped, and the corresponding `BSgenome` package name, and the package to be installed. You can check what BSgenome packages are available on BioConductor via `BSgenome::available.genomes()`, or make your own if necessary (`vignette("BSgenomeForge", package = "BSgenome")`).
+* Whether the aligned reads are paired-end sequencing or not. If single-end sequencing has been used, you need to know the expected DNA fragment length from an experimental measurement, to extend all the reads to this length.
+* An expected fragment length and SD of your DNA fragment size distribution. There are a range of tools that can be used to visualise this on your bam files, for instance [QualiMap](http://qualimap.conesalab.org/), but a rough ballpark should also be experimentally known based on how the DNA material has been processed for sequencing. 
+* What window size to use. In making a `qseaSet`, the genome is split into fixed windows with equal size, and all analysis is performed on these windows as independent units. We generally use 300bp windows, as that is a little bigger than our average DNA fragment sizes, which are in the 150-250bp range. If the DNA is particularly long fragments, a larger bin size may make sense.
+
+For our hg19 example data, we set `BSgenome = "BSgenome.Hsapiens.UCSC.hg19"`, specify the three chromosomes included in the test dataset and give some approximate fragment length and SD values.
+
+```{r, eval = requireNamespace("BSgenome.Hsapiens.UCSC.hg19")}
+makeQset(sampleTable,
+   BSgenome = "BSgenome.Hsapiens.UCSC.hg19",
+   chrSelect = paste0("chr",20:22),
+   windowSize = 300,
+   fragmentLength = 200,
+   fragmentSD = 50,
+   CNVmethod = "none")
+```
+
+A range of messages are produced as output during this process. One thing to note is that by default the process is performed without parallelisation, this can be controlled via the `setMesaParallel` function and will vastly speed up the process when multiple cores are available (but use more RAM).
```

Comment:
Are there any figures for a single-sample usage, even approximate? IOW, how much RAM would a "regular" WGS sample take?

Reply:
To be honest, I never really tested how much RAM was being used.

Reply:
For the record, for 126M reads 45G are *not* enough. But that's far more than usual.